File: vipclaw-admin/src/main/java/com/vipclaw/admin/mapper/SysTokenBlacklistMapper.java
package com.vipclaw.admin.mapper;

import com.baomidou.mybatisplus.core.mapper.BaseMapper;
import com.vipclaw.admin.entity.SysTokenBlacklist;
import org.apache.ibatis.annotations.Mapper;
import org.apache.ibatis.annotations.Param;
import org.apache.ibatis.annotations.Select;

/**
 * Token 黑名单 Mapper
 */
@Mapper
public interface SysTokenBlacklistMapper extends BaseMapper<SysTokenBlacklist> {

    /**
     * 根据 Token 哈希值查询
     * 
     * @param tokenHash Token 哈希值
     * @return Token 黑名单记录
     */
    @Select("SELECT * FROM sys_token_blacklist WHERE token_hash = #{tokenHash} AND expire_time > NOW()")
   SysTokenBlacklist selectByTokenHash(@Param("tokenHash") String tokenHash);
}

File: vipclaw-admin/src/main/kotlin/com/vipamp/vipclaw/admin/mapper/SysTokenBlacklistMapper.kt
package com.vipamp.vipclaw.admin.mapper

import com.vipamp.vipclaw.admin.entity.SysTokenBlacklist
import org.apache.ibatis.annotations.Mapper
import org.apache.ibatis.annotations.Param

/**
 * SysTokenBlacklist Mapper interface
 */
@Mapper
interface SysTokenBlacklistMapper {

    // ==================== Basic CRUD Methods ====================

    fun selectById(@Param("id") id: Long): SysTokenBlacklist?

    fun insert(systokenblacklist: SysTokenBlacklist): Int

    fun updateById(systokenblacklist: SysTokenBlacklist): Int

    fun deleteById(@Param("id") id: Long): Int

    // ==================== Custom Query Methods ====================
    /**
     * Query by Token Hash
     *
     * @param tokenHash Token hash value
     * @return Token blacklist record
     */
    fun selectByTokenHash(@Param("tokenHash") tokenHash: String?): SysTokenBlacklist?
}
